### PR TITLE
Don't emit `doc_markdown` lint for missing backticks if it's inside a quote

### LINKED ISSUE #10262: `doc_markdown` lint should retreat when citing
### Summary

When citing a source in your doc comment, the source may not be written properly. But as you should know, citations may not be changed without mentioning it. And mentioning changes like "emphasis mine", or whatever you'd write when adding backticks and parantheses, can lead to bloat that you may want to prevent.

So, getting `doc_markdown` warnings inside citations isn't desirable and should be prevented on Clippy's side. Finding out what text is inside `"` or `'` may not produce results accurate enough (the latter also being used for apostrophes). So, maybe Clippy should only pay attention to technical markup for this purpose, namely `<q>...</q>` tags and `>` at the start of lines.

### Lint Name

doc_markdown

### Reproducer

**I tried this code:**

```rust
/// AviSynth+ documentation: "If several scripts are open at once and more than one of them loads this plugin, AvisynthPluginInit3 may be called more than once with different IScriptEnvironments." (<https://avisynthplus.readthedocs.io/en/latest/avisynthdoc/FilterSDK/InvertNeg.html>)
pub type InitFnTest1 = fn(foo: i32) -> Result<(), String>;

/// AviSynth+ documentation: <q>If several scripts are open at once and more than one of them loads this plugin, AvisynthPluginInit3 may be called more than once with different IScriptEnvironments.</q> (<https://avisynthplus.readthedocs.io/en/latest/avisynthdoc/FilterSDK/InvertNeg.html>)
pub type InitFnTest2 = fn(foo: i32) -> Result<(), String>;

/// AviSynth+ documentation:
///
/// > If several scripts are open at once and more than one of them loads this plugin, AvisynthPluginInit3 may be called more than once with different IScriptEnvironments.
/// >
/// > &mdash;<https://avisynthplus.readthedocs.io/en/latest/avisynthdoc/FilterSDK/InvertNeg.html>
pub type InitFnTest3 = fn(foo: i32) -> Result<(), String>;
```

**I saw this happen:**

```
warning: item in documentation is missing backticks
   --> [...].rs:258:178
    |
258 | ...hPluginInit3 may be called more than once with different IScriptEnvironments.</q> (<https://avisynthplus.readthedocs.io/en/latest/avis...
    |                                                             ^^^^^^^^^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_markdown
help: try
    |
258 | /// AviSynth+ documentation: <q>If several scripts are open at once and more than one of them loads this plugin, AvisynthPluginInit3 may be called more than once with different `IScriptEnvironments`.</q> (<https://avisynthplus.readthedocs.io/en/latest/avisynthdoc/FilterSDK/InvertNeg.html>)
    |                                                                                                                                                                                  ~~~~~~~~~~~~~~~~~~~~~
```

"AvisynthPluginInit3" and "IScriptEnvironments" have yellow wavy underlining. Written properly, they would need to be "`AvisynthPluginInit3()`" and "`IScriptEnvironment`s"; but this isn't to be changed, as I described above. In light of this, recommending "`IScriptEnvironments`" is especially wrong.

**I expected to see this happen:**

There should be no yellow wavy underlining inside `<q>` and `>` quotes.

### Version

```text
rustc 1.67.0 (fc594f156 2023-01-24)
binary: rustc
commit-hash: fc594f15669680fa70d255faec3ca3fb507c3405
commit-date: 2023-01-24
host: x86_64-pc-windows-msvc
release: 1.67.0
LLVM version: 15.0.6
```

### Additional Labels

_No response_

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #10262.

changelog: Don't emit `doc_markdown` lint for missing backticks if it's inside a quote

### CODE REVIEW

File: clippy_lints/src/doc/markdown.rs [hunk 1]
```diff
@@ -97,7 +104,7 @@ fn check_word(cx: &LateContext<'_>, word: &str, span: Span, code_level: isize) {
     }
 
     // We assume that mixed-case words are not meant to be put inside backticks. (Issue #2343)
```
Comment (blyxyas) [inline clippy_lints/src/doc/markdown.rs:106]: Could you add a comment like this one, but mentioning this issue? Meow meow :cat:
```suggestion
    // We assume that mixed-case words are not meant to be put inside backticks. (Issue #2343)
    // We also assume that quotations should not be changed from the original source, so we will not lint in those. (Issue #10262)
```

**Review by blyxyas — CHANGES_REQUESTED** [review]
Just a mini comment. Does this work with `<q>` and `<quote>`? It would be nice to have that, but it's not a requirement. Meow meow :sparkles:

**Comment by GuillaumeGomez** [comment]
> Just a mini comment. Does this work with `<q>` and `<quote>`? It would be nice to have that, but it's not a requirement. Meow meow ✨

It doesn't, I'll extend the lint to support it.

File: clippy_lints/src/doc/markdown.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply clippy_lints/src/doc/markdown.rs:106]: Good idea! Done.

**Comment by GuillaumeGomez** [comment]
I extended support for `<blockquote>` and `<q>` HTML tags (there is no `<quote>` tag) and added the missing comment.

**Review by blyxyas — APPROVED** [review]
LGTM, thanks! :heart:

**Comment by GuillaumeGomez** [comment]
You forgot to r+ it. 😆

**Comment by blyxyas** [comment]
I was giving it the post-approval pass. 6th pass of my review process.


bors meow meow

**Comment by GuillaumeGomez** [comment]
What is the post-approval pass?

**Comment by blyxyas** [comment]
The post approval pass is the "Wait lemme just re-read the PR just before merging again". It's one of the steps of my book *The review process and its intricacies*

You have 11 reviewing steps:
1. Check the issue that it's solving: "What is this for again?"
2. Check for the surface-level errors.
3. Yeah, I think this PR is good
4. If that day you're feeling extra-helpful, read the PR line by line and check if you can see any logical errors.
5. Approve it.
6. Re-read the PR again.
7. Re-read the PR again. We don't want to merge a typo into the codebase.
8. Re-read the PR again.
9. Merge! :sparkles: :heart:
10. An issue has been raised and someone pinged you. Suffer from impostor syndrome.
11. Repeat.

**Comment by GuillaumeGomez** [comment]
Wow, maybe I should follow your example on this. :laughing:

**Pull request merged**